src/modules/resource/inventory.c

```diff
@@ -626,23 +626,26 @@ static int start_resource_watch (struct inventory *inv,
             flux_log (h, LOG_ERR, "failed to save expiration from R");
     }
 
+    /* Always watch for R updates even with rediscover=true in order to
+     * support instance expiration updates
+     */
+    if (flux_future_then (f, -1., R_update_cb, inv) < 0)
+            flux_log (h, LOG_ERR, "Failed to register callback for R updates");
+
     /* If R == NULL (no conversion possible) or rediscover == true, then
      * we will fall through to dynamic discovery.
      */
     if (R && !config->rediscover) {
         if (inventory_put (inv, R, "job-info") < 0)
             goto done;
-        if (flux_future_then (f,
-                              -1.,
-                              R_update_cb,
-                              inv) < 0)
-            flux_log (h, LOG_ERR, "Failed to register callback for R updates");
-    }
+   }
```

Comment:
I believe a space was accidentally deleted here.

Reply:
Thanks for catching that! I just force-pushed a fix.